### PR TITLE
mihomo 1.18.5 (new formula)

### PR DESCRIPTION
**Pull request by guodong000**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/m/mihomo.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    system "go", "build", "-tags", "with_gvisor", *std_go_args(ldflags:)
+  end
+
+  def post_install
+    (etc/"mihomo").mkpath
+    (etc/"mihomo/config.yaml").write <<~EOS
+      # Document: https://wiki.metacubex.one/config/
+      mixed-port: 7890
+    EOS
+  end
```
Comment (carlocab) [inline Formula/m/mihomo.rb:25]:
```suggestion

    (buildpath/"config.yaml").write <<~EOS
      # Document: https://wiki.metacubex.one/config/
      mixed-port: 7890
    EOS
  end
  pkgetc.install "config.yaml"
```

File: Formula/m/mihomo.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  def caveats
+    <<~EOS
+      You need to customize #{etc}/mihomo/config.yaml.
+
+      mihomo can run without sudo if you don't need to enable TUN.
+
+      Can be used with Proxifier to replace TUN.
```
Comment (carlocab) [inline Formula/m/mihomo.rb:33]:
```suggestion
```
This part isn't specific to Homebrew.

**Review by carlocab — COMMENTED** [review]
Thanks; a few suggestions.

File: Formula/m/mihomo.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    working_dir etc/"mihomo"
+    log_path var/"log/mihomo.log"
+    error_log_path var/"log/mihomo.log"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/mihomo -v")
+
+    (testpath/"mihomo/config.yaml").write <<~EOS
+      mixed-port: 7890
```
Comment (carlocab) [inline Formula/m/mihomo.rb:49]:
```suggestion
      mixed-port: #{free_port}
```

Comment (guodong000) [reply Formula/m/mihomo.rb]: Thank you very much. I have modified it according to your suggestion

**Review by carlocab — APPROVED** [review]
Thanks!

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**